specs/flow_defs_reorganization.adoc

```diff
@@ -0,0 +1,287 @@
+= Flow Framework Refactoring
+
+== Introduction
+
+Current working procedure:
+
+* Any service/agent defining and implementing a flow pushes the corresponding
+  flow definitions to its specific definitions etcd directory.
+* The Agent/service flow framework reads its specific definition files and
+  does all validations and executes the flow.
+
+ex:
+
+* Node-agent pushes tendrl_definitions_node_agent to
+  /tendrl_definitions_node_agent in etcd, gluster integration pushes
+  tendrl_definitions_gluster_integration to
+  /tendrl_definitions_gluster_integration and so on...
+* node-agent is a service that reads flow definitions from
+  /tendrl_definitions_node_agent in etcd and gluster-integration is a process
+  that reads /tendrl_definitions_gluster_integration from etcd and "currently"
+  each of them replicates the flow definition parsing logic.
+
+== Problem description
+
+Any component requiring to define its own flows will then need to implement
+(currently followed)/extend(from a generic one in tendrl/common) the flow
+framework and this requires the component to be run as a service as the flow
+framework in each of these component will only and always read only its
+specific definition and whenever there is a job (intended for it) in the etcd
+queue.
+
+== Use Cases
+
+* Monitoring code cannot be part of any component of tendrl core stack.
+  The monitoring stack needs to provide a flow for configuring the monitoring
+  agents maintained by tendrl.
+
+== Proposed change
+
+* The node-agent pushes a master definition /tendrl_definitions to etcd which
```

Comment:
We should have a namespace `tendrl` in etcd, under which no directory or file needs to have the `tendrl` prefix.